sdk/eventhub/eventhubs-checkpoint-blob/src/blobPartitionManager.ts

```diff
@@ -0,0 +1,209 @@
+  private _containerClient: ContainerClient;
+
+  constructor(containerClient: ContainerClient) {
+    this._containerClient = containerClient;
+  }
+  /**
+   * Get the list of all existing partition ownership from the underlying data store. Could return empty
+   * results if there are is no existing ownership information.
+   *
+   * @param eventHubName The event hub name.
+   * @param consumerGroupName The consumer group name.
+   * @return Partition ownership details of all the partitions that have/had an owner..
+   */
+  async listOwnership(
+    eventHubName: string,
+    consumerGroupName: string
+  ): Promise<PartitionOwnership[]> {
+    const partitionOwnershipArray: PartitionOwnership[] = [];
+    try {
+      for await (const blob of this._containerClient.listBlobsFlat({
+        include: ["metadata"],
+        prefix: `${eventHubName}/${consumerGroupName}/`
+      })) {
+        const blobPath = blob.name.split("/");
+        const blobName = blobPath[blobPath.length - 1];
+        const partitionOwnership: PartitionOwnership = {
+          eventHubName,
+          consumerGroupName,
+          ownerId: blob.metadata!.ownerid,
+          partitionId: blobName,
+          offset: blob.metadata && blob.metadata.offset ? parseInt(blob.metadata.offset) : -1,
+          sequenceNumber:
+            blob.metadata && blob.metadata.sequencenumber
+              ? parseInt(blob.metadata.sequencenumber)
+              : -1,
+          lastModifiedTimeInMS:
+            blob.properties.lastModified && Date.parse(blob.properties.lastModified.toISOString()),
+          eTag: blob.properties.etag,
+          ownerLevel: 0 // this needs to be removed from eventhubs
+        };
+        partitionOwnershipArray.push(partitionOwnership);
+      }
+      return partitionOwnershipArray;
+    } catch (err) {
+      log.error(`Error ocuured while fetching the list of blobs.`, err);
+      throw new Error(`Error ocuured while fetching the list of blobs. \n${err}`);
+    }
+  }
+
+  /**
+   * Claim ownership of a list of partitions. This will return the list of partitions that were owned
+   * successfully.
+   *
+   * @param partitionOwnership The list of partition ownership this instance is claiming to own.
+   * @return A list partitions this instance successfully claimed ownership.
+   */
+  async claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]> {
+    let partitionOwnershipArray: PartitionOwnership[] = [];
+    for (const ownership of partitionOwnership) {
+      const blobName = `${ownership.eventHubName}/${ownership.consumerGroupName}/${ownership.partitionId}`;
+      try {
+        let updatedBlobResponse;
+        const blobClient = this._containerClient.getBlobClient(blobName);
+        if (ownership.eTag) {
+          updatedBlobResponse = await blobClient.setMetadata(
+            {
+              OwnerId: ownership.ownerId ? ownership.ownerId : "",
```

Comment:
This is a case when Event processor instance calls `close()` , then this ownerId will be set as empty. It means that this is partition is available for other instances.

We are not calling `claimOwnership()` in event processor `close()` method now but we will call this method in future, otherwise other processors have to wait till partition ownership expiration time.

Reply:
Gotcha, thanks for the explanation.